docs/NEW_LANGUAGE_FEATURE_INTEGRATION_PROCESS.md

```diff
@@ -0,0 +1,144 @@
+# New Language Feature Check Integration Process [WIP]
+
+This document outlines the procedures for integrating new language features into Checkstyle.
+It includes guidelines for how to discover new checks and
+existing checks that need updates to ensure that Checkstyle remains
+aligned with language enhancements.
+
+## Check Update Procedure
```

Comment:
We will probably expand this section later, but I am not really seeing a procedure here, just more "think abouts" and "reads" this.

I think I will look forward to us using this somewhere and then have this document reference it as a "good" example and things will grow from there. Some of this is just my thoughts when reading this.

`Identify existing tokens` doesn't say how to go about this identification, it seems like it is assumed we are talking about tokens at the same level. I remember seeing something in other issues where we did token scanning and finding checks that had specific tokens.

I have another section on JEPs.

I would love for some sections to contain more specifics like some basic tools for `Other Static Analysis Tools`.

Reply:
>Identify existing tokens doesn't say how to go about this identification, it seems like it is assumed we are talking about tokens at the same level. I remember seeing something in other issues where we did token scanning and finding checks that had specific tokens.

I think this can stay sort of abstract, and we can refine later if we need to. I think that the RECORD_DEF example does a good job of illustrating what we mean here.

Reply:
>I would love for some sections to contain more specifics like some basic tools for Other Static Analysis Tools.

Yes, let's make a list with links to a few of the popular ones, like Intellij Inspections, PMD, and Sonarqube.

Reply:
done